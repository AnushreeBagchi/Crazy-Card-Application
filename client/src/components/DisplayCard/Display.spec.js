import React from "react";
import { shallow } from "enzyme";
import DisplayCard from "./DisplayCard.jsx";

describe("Tests for Display Card", () => {
    function tick() {
        return new Promise(resolve => {
          setTimeout(resolve, 0);
        })
      }
  const mockCard = {
    name: "Student Life Card",
    image: "https://ddialhyn49dxu.cloudfront.net/cards/hsbc2018-large.png",
    apr: "18.9%",
    balanceOfferDuration: "0 months",
    purchaseOfferDuration: "6 months",
    credit: "Â£1200",
    min_income: 0,
    empStatus: "Student",
  };
  it("Renders correctly", () => {
    const wrapper = shallow(<DisplayCard card={mockCard} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("The button should have test ADD Card", () => {
    const onButtonClickMock = jest.fn();
    const wrapper = shallow(
      <DisplayCard card={mockCard} hideAddBtn={false} hideRemoveBtn={true} />
    );
    const addBtn = wrapper.find(".add-btn");
    expect(addBtn.text()).toEqual("ADD Card");
    
  });

  it("On Clicking Add Card, the button should toggle",()=>{
    const onButtonClickMock = jest.fn();
    const wrapper = shallow(
      <DisplayCard card={mockCard} hideAddBtn={false} hideRemoveBtn={true} />
    );
    const addBtn = wrapper.find(".add-btn");
    addBtn.simulate("click");
    jest.useFakeTimers();
    setTimeout(()=>{
        const removeBtn = wrapper.find(".remove-btn");
        jest.runAllTimers();
        expect(removeBtn.text()).toEqual("Remove Card");
    },1000);
  });
});
