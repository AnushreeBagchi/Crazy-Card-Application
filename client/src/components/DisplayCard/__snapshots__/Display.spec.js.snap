// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Display Card Renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  elevation={3}
  key="Student Life Card"
>
  <WithStyles(ForwardRef(CardMedia))
    className="makeStyles-media-2"
    image="https://ddialhyn49dxu.cloudfront.net/cards/hsbc2018-large.png"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      <center>
        Student Life Card
      </center>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Divider))
      variant="middle"
    />
    <WithStyles(ForwardRef(List))
      aria-label="main mailbox folders"
      component="nav"
    >
      <WithStyles(ForwardRef(ListItem))
        key="apr"
      >
        <WithStyles(ForwardRef(ListItemText))
          secondary={
            <React.Fragment>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textPrimary"
                component="span"
                variant="body2"
              >
                Representative % APR (variable)
              </WithStyles(ForwardRef(Typography))>
              <br />
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textSecondary"
                component="span"
                variant="body2"
              >
                18.9%
              </WithStyles(ForwardRef(Typography))>
            </React.Fragment>
          }
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider))
        component="li"
        key="divider-apr"
      />
      <WithStyles(ForwardRef(ListItem))
        key="balanceOfferDuration"
      >
        <WithStyles(ForwardRef(ListItemText))
          secondary={
            <React.Fragment>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textPrimary"
                component="span"
                variant="body2"
              >
                Balance Transfer Offer Duration
              </WithStyles(ForwardRef(Typography))>
              <br />
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textSecondary"
                component="span"
                variant="body2"
              >
                0 months
              </WithStyles(ForwardRef(Typography))>
            </React.Fragment>
          }
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider))
        component="li"
        key="divider-balanceOfferDuration"
      />
      <WithStyles(ForwardRef(ListItem))
        key="purchaseOfferDuration"
      >
        <WithStyles(ForwardRef(ListItemText))
          secondary={
            <React.Fragment>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textPrimary"
                component="span"
                variant="body2"
              >
                Purchase Offer Duration
              </WithStyles(ForwardRef(Typography))>
              <br />
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textSecondary"
                component="span"
                variant="body2"
              >
                6 months
              </WithStyles(ForwardRef(Typography))>
            </React.Fragment>
          }
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider))
        component="li"
        key="divider-purchaseOfferDuration"
      />
      <WithStyles(ForwardRef(ListItem))
        key="credit"
      >
        <WithStyles(ForwardRef(ListItemText))
          secondary={
            <React.Fragment>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textPrimary"
                component="span"
                variant="body2"
              >
                Credit Available
              </WithStyles(ForwardRef(Typography))>
              <br />
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-inline-3"
                color="textSecondary"
                component="span"
                variant="body2"
              >
                Â£1200
              </WithStyles(ForwardRef(Typography))>
            </React.Fragment>
          }
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider))
        component="li"
        key="divider-credit"
      />
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      size="medium"
      variant="contained"
    >
      ADD Card
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="default"
      size="medium"
      variant="contained"
    >
      Remove Card
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
